### ISSUE TITLE
Freeze when try to access settings or download maps

### ISSUE THREAD

**Issue by Uriel6575** [mapsme/omim#1323]
It started in some of the last updates. Map's rendered fine, you can scroll it and even add markers. But when you try to interact with menus (search, settings) somehow, everything becomes freezed.
I'm using Fly IQ4403 with Android 4.2.2. I've played with apktool a little and managed to get a stack trace.

```
  at com.mapswithme.maps.MapFragment.nativeDestroyEngine(Native Method)	
  at com.mapswithme.maps.MapFragment.surfaceDestroyed(MapFragment.java:1043)	
  at android.view.SurfaceView.updateWindow(SurfaceView.java:601)	
  at android.view.SurfaceView.onWindowVisibilityChanged(SurfaceView.java:240)	
  at android.view.View.dispatchWindowVisibilityChanged(View.java:7566)	
  at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1044)	
  at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1044)	
  at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1044)	
  at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1044)	
  at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1044)	
  at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1044)	
  at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1044)	
  at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1383)	
  at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1139)	
  at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:4872)	
  at android.view.Choreographer$CallbackRecord.run(Choreographer.java:776)	
  at android.view.Choreographer.doCallbacks(Choreographer.java:579)	
  at android.view.Choreographer.doFrame(Choreographer.java:548)	
  at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:762)	
  at android.os.Handler.handleCallback(Handler.java:800)	
  at android.os.Handler.dispatchMessage(Handler.java:100)	
  at android.os.Looper.loop(Looper.java:194)	
  at android.app.ActivityThread.main(ActivityThread.java:5410)	
  at java.lang.reflect.Method.invokeNative(Native Method)	
  at java.lang.reflect.Method.invoke(Method.java:525)	
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)	
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)	
  at dalvik.system.NativeStart.main(Native Method)	
```

**Comment by deathbaba** [comment]
Is it a market version? Can you please share logcat logs?

**Comment by Uriel6575** [comment]
Yes, it is from market.

[log.txt](https://github.com/mapsme/omim/files/86268/log.txt)

**Comment by deathbaba** [comment]
There is a native crash in the log in libc stacktrace, but it is not clear what has caused it. It can be connected with our new graphics engine (old versions worked fine, right?), but @rokuz says he can't debug the issue without a device in his hands.

**Comment by nyqvist** [comment]
I have the same issue with the market version. I use Sony Xperia. Old version worked fine for me.

**Comment by deathbaba** [comment]
@nyqvist @Uriel6575 can you please install this debug version from master branch and send us logs after reproducing these crashes?
https://www.dropbox.com/s/vh15a8svyaw5ny5/android-google-universal-debug-5.5-160114.apk?dl=1

CC @darina @rokuz

**Comment by Uriel6575** [comment]
This time I couldn't have reproduced the crash - it simply freeze every time.
[log.txt](https://github.com/mapsme/omim/files/90322/log.txt)

**Comment by nyqvist** [comment]
When I open the debug version after installing I  see a message in red text that says that the download failed. Same thing when pressing retry button.

**Comment by Uriel6575** [comment]
If fixed for me in 5.2.6 apk from your web site. But you have to wait a second before interacting with UI. Otherwise it would freeze like before.
